ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogBloomCacher.java

```diff
@@ -202,17 +206,20 @@ private boolean populateLatestSegment(final long eventBlockNumber) {
     return false;
   }
 
-  private void ensurePreviousSegmentsArePresent(final long blockNumber) {
+  public void ensurePreviousSegmentsArePresent(
+      final long blockNumber, final boolean overrideCacheCheck) {
     if (!cachingStatus.isCaching()) {
       scheduler.scheduleFutureTask(
           () -> {
             long currentSegment = (blockNumber / BLOCKS_PER_BLOOM_CACHE) - 1;
             while (currentSegment > 0) {
               try {
-                if (!cachedSegments.getOrDefault(currentSegment, false)) {
+                if (overrideCacheCheck || !cachedSegments.getOrDefault(currentSegment, false)) {
```

Comment:
@shemnon Would it be alright to change `cachedSegments` into a `TreeSet`? Couldn't see anywhere we need the model the ternary of <true, false, not present>

Reply:
@RatanRSur This may be an option but let's leave refactoring to another story.

Reply:
In another PR is fine.

Reply:
👍